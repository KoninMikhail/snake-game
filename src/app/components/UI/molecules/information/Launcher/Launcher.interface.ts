import { CSSProperties } from 'react';
import { IImage } from '@ui/atoms/images/images.interface';

export interface ILauncher {
    image: IImage;
    message: string;
    style?: CSSProperties;
}
