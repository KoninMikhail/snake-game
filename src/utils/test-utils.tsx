import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
