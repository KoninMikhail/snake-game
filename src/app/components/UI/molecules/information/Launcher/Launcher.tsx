import {
    StyledStartBox,
    StyledStartBoxImageContainer,
    StyledStartBoxMessageContainer,
    StyledStopBoxContentHolder,
} from './style/StyledStartBox';
import { FitWidthImage } from '@ui/atoms/images/FitWidthImage/FitWidthImage';
import { ILauncher } from '@ui/molecules/information/Launcher/Launcher.interface';
import { LightTextShadow } from '@ui/atoms/assets/shadows/text/LightTextShadow/LightTextShadow';
import { Caption } from '@ui/atoms/typography/caption/Caption/Caption';
import { BlinkAnimation } from '@ui/atoms/assets/animations/BlinkAnimation/BlinkAnimation';

/**
 * @name Launcher
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for show the start box state home screen
 *
 * @param image - image for output
 * @param message - message for print
 * @param style - css styles
 */
export const Launcher = ({ image, message, style }: ILauncher) => {
    return (
        <StyledStartBox style={style}>
            <StyledStopBoxContentHolder>
                <StyledStartBoxImageContainer>
                    <FitWidthImage {...image} />
                </StyledStartBoxImageContainer>
                <StyledStartBoxMessageContainer>
                    <BlinkAnimation>
                        <LightTextShadow>
                            <Caption align={'center'}>{message}</Caption>
                        </LightTextShadow>
                    </BlinkAnimation>
                </StyledStartBoxMessageContainer>
            </StyledStopBoxContentHolder>
        </StyledStartBox>
    );
};
