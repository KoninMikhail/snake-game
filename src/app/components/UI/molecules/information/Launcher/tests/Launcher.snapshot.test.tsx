import { screen } from '@testing-library/react';
import { renderWithProviders } from '@utils/test-utils';
import { Launcher } from '@ui/molecules/information/Launcher/Launcher';

describe('Launcher', () => {
    const content = 'test message';

    it('should equal snapshot', () => {
        renderWithProviders(<Launcher image={{ sourceID: 'background' }} message={content} />);
        expect(screen.getByText(content)).toMatchSnapshot();
    });
});
