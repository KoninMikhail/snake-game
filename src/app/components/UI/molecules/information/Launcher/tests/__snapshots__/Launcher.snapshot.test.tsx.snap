// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launcher should equal snapshot 1`] = `
.c0 {
  font-size: 1.3em;
  font-weight: normal;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 1.18em;
  }
}

<span
  class="c0"
  style="text-shadow: 2px 2px 0 rgba(0, 0, 0, 1);"
>
  test message
</span>
`;
